tests/drivers/test_json.py

```diff
@@ -0,0 +1,75 @@
+import json
+import pytest
+
+from dlme_airflow.drivers.json import JsonSource
+
+
+def test_happy_path(requests_mock):
+    # mock the http call to return some real loc.gov data
+    loc_data = json.load(open("tests/data/json/loc.json"))
+    requests_mock.get(
+        "https://www.loc.gov/collections/persian-language-rare-materials/?c=100&fo=json",
+        json=loc_data,
+    )
+
+    # the collection url defined in the Intake catalog
+    collection_url = (
+        "https://www.loc.gov/collections/persian-language-rare-materials/?c=100&fo=json"
+    )
+
+    # this jmespath configuration is required in the intake catalog
+    metadata = {
+        "record_selector": "content.results",
+        "fields": {
+            "id": {"path": "id"},
+            "title": {"path": "title"},
+            "thumbnail": {
+                "path": "resources[0].image",
+                "optional": True,  # some records lack a thumbnail
+            },
+        },
+    }
+
+    # create our JsonSource object
+    js = JsonSource(collection_url, metadata=metadata)
+
+    # get the DataFrame
+    df = js.read()
+
+    assert len(df) == 100
+    assert len(df.columns) == 3
+    assert set(df.columns) == set(["id", "title", "thumbnail"])
```

Comment:
now i also want to update the expected columns comparison tests in `test_iiif_json.py` and `test_xml.py` with this wrapping-in-a-`set` approach 🙂   way more concise/readable.